teachertool/src/components/CriteriaInstanceDisplay.tsx

```diff
@@ -22,22 +23,32 @@ const InlineInputSegment: React.FC<InlineInputSegmentProps> = ({
     shouldExpand,
     numeric,
 }) => {
-    function onChange(newValue: string) {
+    const [isEmpty, setIsEmpty] = useState(!initialValue);
+
+    function onDebouncedChange(newValue: string) {
```

Comment:
I understand the want to have the debounce to limit the amount of state changes, but I'm not sure about introducing and ability for something to trigger on a debounce and something else fires immediately. It feels like an anti-pattern to have logic on a debounce component that doesn't get "debounced". Maybe just setting the `intervalMs` lower than 500ms could be a solution?

Reply:
I think even 500ms is going to be noticeable. I'm not sure I agree with the idea that it's an anti-pattern. The debounce is purely internal, not really relevant to the user, but the visual feedback of an error is something they see and interact with. After all, we don't debounce the characters appearing in the input field, and I think this error display is closer to that scenario than it is to us saving the input internally (which is what gets debounced).

Reply:
I'm meaning anti-pattern as a code smell. Having the props `onChange` and `onDebouncedChange` are a bit hard to understand. If I'm importing a debounced input, I'm expecting all of the things that I input to be debounced, so if I use `onChange` and it doesn't get debounced, I would consider it a bug. 

I agree that the error is different. In this scenario, we're just checking whether or not the input is empty whenever the input is interacted with. "onChange" is a bit misleading because it could be that no text was changed, that the box was just clicked on.

I propose getting rid of `onDebouncedChange`, keeping `onChange` as it was, and adding a prop like `checkIsEmpty` or something and that logic can be added to that if statement that you added under the `setTimeout` in DebouncedInput.

Reply:
In this scenario it's checking isEmpty, but there could be scenarios where it's more involved (like putting a letter in a number field, exceeding some length, or even reaching some minimum length to enter a "valid" state instead of an error one). I don't think we want to make the component that situationally specific.

If this is a sticking point, I would sooner just get rid of the debouncing altogether and have it react in real time.

Reply:
I think checkInputIsValid is probably still too specific and would imply that the DebouncedInput component itself is going to have some styling / reaction to the invalid state.

Would it be preferable if we had `onChange` which is debounced and `onChangeImmediate` which is not? That feels less intuitive to me personally, but perhaps works as a compromise?

Reply:
On second thought, I think I'll just remove the debouncing for now.

Reply:
On second thought, I think I will just remove the debounce.